add "Lygisk vs Magisk"  in README.md (#22)
programminghoch10/Lygisk: README.md

It's the same way for switching back to Magisk.

## Lygisk vs Magisk

Lygisk has been developed for one specific case:
* `A-only` device
* `/data` is encrypted
* the recovery cannot decrypt `/data`
* the recovery is `addon.d` capable
This is the case for every LineageOS device running the LineageOS recovery.

If you patch `boot.img` or you have `/data` not encrypted, use Magisk.

> If you have an `addon.d`-compatible ROM (such as LineageOS) with encrypted `/data`, use Lygisk.

Author: ```suggestion
If you have an `addon.d`-compatible recovery (such as LineageOS recovery) with encrypted `/data`, use Lygisk.
```

It's the same way for switching back to Magisk.

## Lygisk vs Magisk

Lygisk has only an advantage that if you have an A-only device that is encrypted and the recovery cannot
(automatically) decrypt `/data`,

> then reinstalling Lygisk during OTA is automatic.

Reviewer: I don't think the ROM has to have any support for `addon.d`, what would the ROM even do with it?
It only matters that the recovery is actually running `addon.d/*.sh` after install.
This is also why I explicitly wrote `running the LineageOS recovery`.

That can also be an advantage on TWRP when your device is encrypted because TWRP can't decrypt `/data` during OTA either
(I believe, I don't use it).

If you patch `boot` or have `/data` not encrypted (that's probably with TWRP as I heard it prevents encryption),
you should use Magisk.
If you have an addon.d-compatible ROM (such as Lineage OS) with encrypted `/data`, use Lygisk.

That's not about keeping the stuff in addon.d up-to-date
but to have everything needed to install is actually in /system/addon.d because Magisk only stores the install-script 
there but all the files needed for that are within /data
(which is encrypted and thus unavailable).

If you update Lygisk (via direct install),
the update will both update Lygisk and the files in `addon.d` to ensure that Lygisk survives OTA.

> 

Author: Yes because until a year ago, this feature did not work.
Also this strategy works for migrating Magisk (debug) -> Lygisk.

> 

Reviewer: When the feature came to live is irrelevant.
And mentioning this in this way does not convey in any way that Magisk debug migration to Lygisk is possible now. I don't think many people do this exact migration anyways.

> 

Author: Yeah, that's true.

That is actually the whole reason why Lygisk exists: You don't need to re-flash Lygisk after OTA manually.

> 

Author: There are people who need this piece of information explicitly

> If you use Lygisk when your storage is not encrypted anyway, it won't give you an advantage, but it will still work.
> 

Author: ```suggestion
If you use Lygisk when your storage is not encrypted anyway, it won't give you an advantage.
```

> If you use Lygisk when your storage is not encrypted anyway, it won't give you an advantage, but it will still work.
> 

Reviewer: ```suggestion
If you use Lygisk with unencrypted `/data`, it won't give you any advantages over Magisk.
```

> If you use Lygisk when your storage is not encrypted anyway, it won't give you an advantage, but it will still work.
> 

Author: ```suggestion
If you use Lygisk with unencrypted `/data`, it won't give you any advantage over Magisk.
```
